determine_needed_tags.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+
+component=$1
+build_version=$2
```

Comment:
Added the comments and a usage info. This has actually lead me to another pitfall. Right now, `make` would fail if we build an image for a new component that is not yet available at dockerhub. 

I'll think about this case a bit more and see how to handle it. 

Please feel free to review the usage info and just suggest/apply changes.